llvm/lib/Transforms/Instrumentation/PGOInstrumentation.cpp

```diff
@@ -1845,14 +1845,19 @@ static bool InstrumentAllFunctions(
   std::unordered_multimap<Comdat *, GlobalValue *> ComdatMembers;
   collectComdatMembers(M, ComdatMembers);
 
+  bool AnythingInstrumented = false;
   for (auto &F : M) {
     if (skipPGOGen(F))
       continue;
     auto &TLI = LookupTLI(F);
     auto *BPI = LookupBPI(F);
     auto *BFI = LookupBFI(F);
     instrumentOneFunc(F, &M, TLI, BPI, BFI, ComdatMembers, IsCS);
+    AnythingInstrumented = true;
   }
+  if (!AnythingInstrumented)
+    return false;
+
   return true;
```

Comment:
```suggestion
  return AnythingInstrumented;
```

Reply:
@ellishg Thank you very much for the suggestion. This is an artifact after splitting the patch. Applied.